llvm/lib/Transforms/IPO/DynUntrustedAllocPost.cpp

```diff
@@ -212,82 +232,89 @@ class DynUntrustedAllocPost : public ModulePass {
 
           auto index = index_iter->second;
 
-          BasicBlock::iterator iter(CS.getInstruction());
-          auto prev_inst = BB->getInstList().getPrevNode(*CS.getInstruction());
+          // Set UniqueID for hook function
+          auto callInst = CS.getInstruction();
           auto id = IDG.getConstID(M);
-
           CS.setArgument(index, id);
+          
           ++total_hooks;
-          LLVM_DEBUG(errs() << "modified callsite:\n");
-          LLVM_DEBUG(errs() << *CS.getInstruction() << "\n");
 
-          if (!prev_inst)
-            continue;
+          if (remove_hooks)
+            hookList.push_back(callInst);
 
-          CallSite CSPrev(prev_inst);
-          if (!CSPrev)
-            continue;
+          // If provided a valid path, modify given instruction
+          if (!mpk_profile_path.empty()) {
+            // Get Call Instr that hook references
+            auto allocFunc = CS.getArgument(0);
+            if (isa<CallInst>(allocFunc) || isa<InvokeInst>(allocFunc)) {
```

Comment:
cant this be something like `if( auto* allocInst =  dyn_cast<CallBase>(allocFunc))`

Reply:
Unfortunately not, since deallocHook does not have a reference to the dealloc instruction, and attempting to cast in the conditional throws an assertion error.

Reply:
Switched to dyn_cast<> as suggested.